js/components/utils.js

```diff
@@ -632,43 +632,89 @@ var Utils = {
    *
    * @function getLanguageToUse
    * @param {Object} skinConfig - The skin configuration file to read languages from
-   * @returns {String} The ISO code of the language to use
+   * @returns {String} The ISO 639-1 code of the language to use or an empty string
    */
   getLanguageToUse: function(skinConfig) {
-    var localization = skinConfig.localization;
-    var language, availableLanguages;
-
-    // set lang to default lang in skin config
-    language = localization.defaultLanguage;
-
-    // if no default lang in skin config check browser lang settings
-    if (!language) {
-      if (window.navigator.languages) {
-        // A String, representing the language version of the browser.
-        // Examples of valid language codes are: "en", "en-US", "de", "fr", etc.
-        language = window.navigator.languages[0];
+    let language = '';
```

Comment:
As for me it's better to return value immediately than use `let` and long `if` clause
like this:
```
getLanguageToUse: function(skinConfig) {
  if (!skinConfig) {
    return '';
  }

  const localization = skinConfig.localization;
  const userBrowserLanguage = this.getUserBrowserLanguage();

  const isLanguageCodeInAvailablelLanguageFile = this.isLanguageCodeInAvailablelLanguageFile(
    localization.availableLanguageFile, userBrowserLanguage
  );

  if (!userBrowserLanguage || !localization || !isLanguageCodeInAvailablelLanguageFile) {
    return this.getDefaultLanguage(localization);
  }

  return userBrowserLanguage;
}
```

Reply:
@ooyala-amohnatkin an interesting catch! Thank you))